initialization/vrf_search.go

```diff
@@ -0,0 +1,183 @@
+		logger.Info("looking for VRF nonce in file", zap.String("file", file.Name()))
+
+		file, err := os.Open(filepath.Join(initOpts.DataDir, file.Name()))
+		if err != nil {
+			logger.Error("failed to open file", zap.Error(err))
+			continue
+		}
+		defer file.Close()
+
+		idx, l, err := searchForNonce(ctx, bufio.NewReader(file), difficulty, woReference)
+		if err != nil {
+			return 0, nil, fmt.Errorf("failed to search for nonce: %w", err)
+		}
+		if l != nil {
+			nonceValue = l
+			nonce = firstLabelIndex + idx
+			logger.Info("found nonce", zap.Uint64("index", nonce), zap.String("label", hex.EncodeToString(nonceValue)))
+			difficulty = nonceValue // override difficulty to the new lowest label
+
+		}
+	}
+	if nonceValue != nil {
+		logger.Info("updating postdata_metadata.json with found nonce", zap.Uint64("nonce", nonce), zap.String("NonceValue", hex.EncodeToString(nonceValue)))
+		metadata.Nonce = &nonce
+		metadata.NonceValue = shared.NonceValue(nonceValue)
+		SaveMetadata(initOpts.DataDir, metadata)
+		return nonce, nonceValue, nil
+	}
+	return 0, nil, ErrNonceNotFound
+}
+
+// searchForNonce searches for a nonce in the given reader.
+func searchForNonce(ctx context.Context, r io.Reader, difficulty []byte, oracle *oracle.WorkOracle) (nonce uint64, nonceValue []byte, err error) {
+	labelBuf := make([]byte, postrs.LabelLength)
+	labelIndex := uint64(0)
+	for {
+		select {
+		case <-ctx.Done():
+			return nonce, labelBuf, ctx.Err()
+		default:
+			// continue looking for a nonce
+		}
+
+		_, err := io.ReadFull(r, labelBuf)
+		switch {
+		case err == io.EOF:
+			return nonce, nonceValue, nil
+		case err == io.ErrUnexpectedEOF:
+			return 0, nil, errors.New("unexpected end of file - file is truncated. please reinit it")
+		}
+
+		ok, err := checkLabel(labelIndex, labelBuf, difficulty, oracle)
+		if err != nil {
+			return 0, nil, fmt.Errorf("checking label: %w", err)
+		}
+		if ok {
+			nonce = labelIndex
+			nonceValue = append(nonceValue[:0], labelBuf...)
+			difficulty = nonceValue // override difficulty to the new lowest label
+		}
+		labelIndex++
+	}
+}
+
+// checkLabels checks if label is lower than difficulty.
+// It will regenerate the whole 32B of the label if its most significant 16B are equal to difficulty
+// in order to check the lower bytes.
+// * index:      the index of the label (used to regenerate it)
+// * label:		 16B most significant bytes of the label
+// * difficulty: all 32b of the difficulty
+// * wo:         the oracle used to regenerate the label.
+func checkLabel(index uint64, label, difficulty []byte, wo *oracle.WorkOracle) (bool, error) {
+	comp := bytes.Compare(label, difficulty)
+	if comp < 0 {
+		return true, nil
+	} else if comp == 0 {
+		// need to regenerate label to verify lower bytes
+		res, err := wo.Position(index)
+		if err != nil {
+			return false, fmt.Errorf("failed to regenerate label: %w", err)
+		}
+		return bytes.Compare(res.Output, difficulty) < 0, nil
+	}
+	return false, nil
```

Comment:
Would it make sense to use switch here?

```suggestion
	switch {
	case comp < 0:
		return true, nil
	case comp == 0:
		// need to regenerate label to verify lower bytes
		res, err := wo.Position(index)
		if err != nil {
			return false, fmt.Errorf("failed to regenerate label: %w", err)
		}
		return bytes.Compare(res.Output, difficulty) < 0, nil
	default: // comp > 0
		return false, nil
	}
```